subliminal/providers/legendastv.py

```diff
@@ -359,8 +375,16 @@ def query(self, language, title, season=None, episode=None, year=None):
                     if not a.pack and 'episode' in guess and guess['episode'] != episode:
                         continue
 
+                # calculates a good expiration time
+                if a.timestamp:
+                    delta = max(0, (datetime.datetime.now() - a.timestamp).total_seconds())
+                    # recently uploaded archives have short expiration time
+                    expiration_time = min(RELEASES_EXPIRATION_TIME, delta / 8)
+                else:
+                    expiration_time = RELEASES_EXPIRATION_TIME
+
                 # attempt to get the releases from the cache
-                releases = region.get(releases_key.format(archive_id=a.id))
+                releases = region.get(releases_key.format(archive_id=a.id), expiration_time=expiration_time)
```

Comment:
I think the max isn't necessary unless we have an upload time in the future which. We also might want some margin of error.
Issues with this will be hard to identify so we better think this through now :p